### PR TITLE
Trim leading and trailing whitespace from search query

### LINKED ISSUE #1621: trim leading and trailing whitespace from search
I just tried searching for "JOROL " (without the double quotes, where are here to show the trailing space), and didn't get his account listed at the top. I had pasted his PAUSE id from an email and accidentally picked up the trailing space.

Deleting the space and searching again resulted in the expected user account.

### PR DESCRIPTION
**Pull request by zakame**
Fixes #1621.

### CODE REVIEW

File: lib/MetaCPAN/Web/Controller/Search.pm [hunk 1]
```diff
@@ -30,6 +30,10 @@ sub index : Path : Args(0) {
         $query =~ s{\.pm\b}{};
     }
 
+    # trim leading/traling whitespace
```
Comment (haarg) [inline lib/MetaCPAN/Web/Controller/Search.pm]: Can this comment be removed or fixed?

Comment (zakame) [inline lib/MetaCPAN/Web/Controller/Search.pm]: Sure

Comment (zakame) [inline lib/MetaCPAN/Web/Controller/Search.pm]: Now removed

**Comment by zakame** [comment]
Thanks!

**Comment by neilb** [comment]
:+1:

**Pull request merged**